From 6ac0853d92052fc40a1be1f4303aebe5011a4166 Mon Sep 17 00:00:00 2001
From: Siva Rajesh <siva.rajesh.jarugula@xilinx.com>
Date: Fri, 3 Feb 2017 00:31:00 +0530
Subject: [PATCH 0229/1851] drm: mipi: Bring DT format description closer to
 the hardware

commit 2a8c32259cc5b1d8fead70afde27f8faa4ab5959 from
https://github.com/Xilinx/linux-xlnx.git

Use the DSI pixel format codes as defined in the include/drm/drm_mipi_dsi.h
to describe data type for pixel format. Based on data type the incoming
pixel stream is converted to byte stream to match with the DSI requirements
detailed in sec 8.8 of the MIPI Alliance Standard for DSI specification.
This gets rid of yet another string-based format description, while
keeping the bindings simple enough and close to the hardware.

Signed-off-by: Siva Rajesh <siva.rajesh.jarugula@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Reviewed-by: Hyun Kwon <hyunk@xilinx.com
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/dt-bindings/drm/mipi-dsi.h | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 include/dt-bindings/drm/mipi-dsi.h

diff --git a/include/dt-bindings/drm/mipi-dsi.h b/include/dt-bindings/drm/mipi-dsi.h
new file mode 100644
index 000000000000..c6f37ec661fe
--- /dev/null
+++ b/include/dt-bindings/drm/mipi-dsi.h
@@ -0,0 +1,11 @@
+#ifndef __DT_BINDINGS_DRM__
+#define __DT_BINDINGS_DRM__
+/*
+ * MIPI DSI pixel formats as defined in the include/drm/drm_mipi_dsi.h"
+ */
+#define MIPI_DSI_FMT_RGB888		0
+#define MIPI_DSI_FMT_RGB666		1
+#define MIPI_DSI_FMT_RGB666_PACKED	2
+#define MIPI_DSI_FMT_RGB565		3
+
+#endif /* _DT_BINDINGS_DRM__ */
-- 
2.31.1

